test/e2e/syncer/syncer_test.go

```diff
@@ -153,14 +154,28 @@ func TestSyncerLifecycle(t *testing.T) {
 		}
 		if err != nil {
 			t.Errorf("saw an error waiting for downstream deployment %s/%s to be created: %v", downstreamNamespaceName, upstreamDeployment.Name, err)
-			return false
-		}
-		// Check for available replicas if downstream is capable of actually running the deployment
-		if len(framework.TestConfig.PClusterKubeconfig()) > 0 && expectedAvailableReplicas != deployment.Status.AvailableReplicas {
-			return false
 		}
 		return true
 	}, wait.ForeverTestTimeout, time.Millisecond*100, "downstream deployment %s/%s was not synced", downstreamNamespaceName, upstreamDeployment.Name)
 
+	t.Logf("Check for available replicas if downstream is capable of actually running the deployment")
+	expectedAvailableReplicas := int32(1)
+	require.Eventually(t, func() bool {
+		deployment, err = downstreamKubeClient.AppsV1().Deployments(downstreamNamespaceName).Get(ctx, upstreamDeployment.Name, metav1.GetOptions{})
+		if err != nil {
+			t.Errorf("saw an error waiting for downstream deployment %s/%s to be created: %v", downstreamNamespaceName, upstreamDeployment.Name, err)
```

Comment:
```suggestion
			t.Errorf("error getting downstream deployment %s/%s: %v", downstreamNamespaceName, upstreamDeployment.Name, err)
```

Reply:
Also, need to `return false`?

Reply:
Should it be a test error we can't get it if the `require.Eventually` is going to retry?

Reply:
We previously successfully retrieved it. Maybe this should just be `require.NoError(t, err)`?

Reply:
done